examples/ensemble/plot_gradient_boosting_quantile.py

```diff
@@ -32,41 +36,87 @@ def f(x):
+gbr_mse = GradientBoostingRegressor(loss='ls', n_estimators=250, max_depth=3,
+                                    learning_rate=.1, min_samples_leaf=9,
+                                    min_samples_split=9)
+
+# %%
+# Let's fit this models and three others trained with
+# the quantile loss and alpha=0.05, 0.5, 0.95.
+# The models obtained for alpha=0.05 and alpha=0.95
+# produce a 95% confidence interval. The model trained
+# with alpha=0.5 produce the median regression:
+# there are the same number of targets above and below
+# the predicted values.
+gbrs = {
+    "q%1.2f" % alpha:
+        GradientBoostingRegressor(loss='quantile', alpha=alpha,
+                                  n_estimators=250, max_depth=3,
+                                  learning_rate=.1, min_samples_leaf=9,
+                                  min_samples_split=9)
+    for alpha in [0.05, 0.5, 0.95]
+}
+gbrs['MSE'] = gbr_mse
+
+for alpha, gbr in gbrs.items():
+    gbr.fit(X_train, y_train)
+
+# %%
+# Let's measure the models given :func:`mean_square_error` and
+# :func:`pinball_loss` metrics on the training datasets.
+results = []
+for name, gbr in sorted(gbrs.items()):
+    metrics = {'model': name}
+    y_pred = gbr.predict(X_train)
+    for alpha in [0.05, 0.5, 0.95]:
+        metrics["pbl=%1.2f" % alpha] = pinball_loss(
+            y_train, y_pred, alpha=alpha)
+    metrics['MSE'] = mean_squared_error(y_train, y_pred)
+    results.append(metrics)
+DataFrame(results).set_index('model')
+
+# %%
+# One column shows all models evaluated by the same metric.
+# The number of the diagonal is the metric evaluated on a
+# model trained to minimize this same metric. It is expected
+# to be the minimum value on this column.
+
+# %%
+# We do the same on the test set.
+results = []
+for name, gbr in sorted(gbrs.items()):
+    metrics = {'model': name}
+    y_pred = gbr.predict(X_test)
+    for alpha in [0.05, 0.5, 0.95]:
+        metrics["pbl=%1.2f" % alpha] = pinball_loss(
+            y_test, y_pred, alpha=alpha)
+    metrics['MSE'] = mean_squared_error(y_test, y_pred)
+    results.append(metrics)
+DataFrame(results).set_index('model')
+
+# %%
+# Let's finally plot the function, the prediction and the
+# 95% confidence interval based on the MSE.
+y_pred = gbrs['MSE'].predict(xx)
+y_lower = gbrs['q0.05'].predict(xx)
+y_upper = gbrs['q0.95'].predict(xx)
+y_med = gbrs['q0.50'].predict(xx)
 
-# Plot the function, the prediction and the 95% confidence interval based on
-# the MSE
 fig = plt.figure()
 plt.plot(xx, f(xx), 'g:', label=r'$f(x) = x\,\sin(x)$')
-plt.plot(X, y, 'b.', markersize=10, label=u'Observations')
-plt.plot(xx, y_pred, 'r-', label=u'Prediction')
+plt.plot(X_test, y_test, 'b.', markersize=10, label='Test observations')
+plt.plot(xx, y_med, 'r-', label='Prediction Median', color="orange")
+plt.plot(xx, y_pred, 'r-', label='Prediction MSE')
 plt.plot(xx, y_upper, 'k-')
 plt.plot(xx, y_lower, 'k-')
 plt.fill(np.concatenate([xx, xx[::-1]]),
```

Comment:
I would rather move the plot up in the notebook, before the train and test metrics computation and tables.